torch/distributed/distributed_c10d.py

```diff
@@ -1064,6 +1064,159 @@ def recv(tensor, src=None, group=None, tag=0):
         return src
 
 
+Tensordtype2ncclDataType_t={
+    # dtype              # corresponding to ncclDataType_t from `nccl.h`
+    "torch.int8"    : 0, # ncclInt8       = 0
+    "torch.uint8"   : 1, # ncclUint8      = 1
+    "torch.int32"   : 2, # ncclInt32      = 2
+                         # ncclUint32     = 3
+    "torch.int64"   : 4, # ncclInt64      = 4
+                         # ncclUint64     = 5
+    "torch.float16" : 6, # ncclFloat16    = 6
+    "torch.float32" : 7, # ncclFloat32    = 7
+    "torch.float64" : 8, # ncclFloat64    = 8
+    "torch.bfloat16": 9, # ncclBfloat16   = 9
+}
```

Comment:
Is there a reason to introduce NCCL specific coding here? Same below.

Reply:
No special reason. Just need a way to encode the `dtype`. For overall consistency of `PyTorch` source code, I choosed to use the NCCL specific coding here.

Reply:
Just need a mapping to encode tensor dtypes. I have changed that to `ScalarType`.